hive-event-listeners/apiary-gluesync-listener/src/main/java/com/expediagroup/apiary/extensions/gluesync/listener/ApiaryGlueSync.java

```diff
@@ -76,8 +82,45 @@ public ApiaryGlueSync(Configuration config, AWSGlue glueClient, String gluePrefi
   }
 
   @Override
-  public void onCreateTable(CreateTableEvent event) throws MetaException {
-    if (event.getStatus() == false) {
+  public void onCreateDatabase(CreateDatabaseEvent event) {
+    if (!event.getStatus()) {
+      return;
+    }
+    Database database = event.getDatabase();
+    try {
+      CreateDatabaseRequest createDatabaseRequest = new CreateDatabaseRequest()
+          .withDatabaseInput(transformDatabase(database));
+      glueClient.createDatabase(createDatabaseRequest);
+      log.debug(database + " database created in glue catalog");
+    } catch (AlreadyExistsException e) {
+      log.debug(database + " database already exists in glue, updating....");
+      UpdateDatabaseRequest updateDatabaseRequest = new UpdateDatabaseRequest()
+          .withName(glueDbName(database.getName()))
+          .withDatabaseInput(transformDatabase(database));
+      glueClient.updateDatabase(updateDatabaseRequest);
+      log.debug(database + " database updated in glue catalog");
+    }
+  }
+
+  @Override
+  public void onDropDatabase(DropDatabaseEvent event) {
+    if (!event.getStatus()) {
+      return;
+    }
+    Database database = event.getDatabase();
+    try {
+      DeleteDatabaseRequest deleteDatabaseRequest = new DeleteDatabaseRequest()
+          .withName(glueDbName(database.getName()));
+      glueClient.deleteDatabase(deleteDatabaseRequest);
+      log.debug(database + " database deleted from glue catalog");
+    } catch (EntityNotFoundException e) {
+      log.debug(database + " database doesn't exist in glue catalog");
```

Comment:
Same here

Reply:
That's true, it is just entity not found... I suppose if the delete/create database fails it logs by default? Since it is not handled I suppose it will log

Reply:
That's something @andreeapad needs to check yes.

Reply:
the metastore will probably log the errors (the upstream code)